libs/datamodel/core/src/ast/reformat/reformatter.rs

```diff
@@ -339,6 +382,16 @@ impl<'a> Reformatter<'a> {
                 _ => Self::reformat_generic_token(target, &current),
             }
         }
+        if let Ok(missing_field_directives) = self.missing_field_directives.as_ref() {
+            for missing_field_directive in missing_field_directives.iter() {
+                if &missing_field_directive.field == field_name {
```

Comment:
I think we are still missing an if for the name of the model here. I think this function should take `model_name` as an additional parameter.

Reply:
yes, I added a test case for that. It did add the directive multiple times.